fgcom-mumble is now a  comprehensive propagation simulator with realistic frequency usage patterns and solar-terrestrial physics modeling.

Supermagnum opened the pull request:
Enhance fgcom-mumble to become a comprehensive amateur radio HF propagation simulator with realistic frequency usage patterns and solar-terrestrial physics modeling.

All tests passes, and the install system works.
Note: there may be undiscovered bugs.

relevant to:
https://github.com/hbeni/fgcom-mumble/issues/228

hbeni commented:
Just did a quick peek at the implementation.
Architecture wise I'm still not sure, if this is going into the right direction. Also, I really dislike the Idea that this makes the codebase explode: The PR makes it go from ~3k lines to 29k. This needs REALLY good and comprehensive checking before a merge, more so as its AI generated. Also, deep tests must be made in the field with real users.
So even if it does what the PR promises, it needs careful checks.
That is probably beyond my capacity currently and for the foreseeable future.

I really like the ideas and proposals and features you brought in. Please don't get me wrong.
However, as the massive change in things this brings up, currently it might be a better option to think about a fork of the project - so this here can remain a relatively small fork, tailored to the needs of the flightsim users, while you can maintain the ultra-version for the radio simulation enthusiasts?
Just thinking out loud and bringing my toughts to the table. Tried to reach you on discord, but cant find you (see issue ticket).

hbeni reviewed (commented):
I guess that is what to expect when trying to let AI overtake humans 😅

hbeni commented:
Maybe you can use the existing fgcom.hallinger.org testserver - not sure if I understood the changes to the server side correctly, if plain mumble is still enough (what is a a project design choice, btw).

I do prefer to discuss things relating to pull requests (especially this large) in public at a linked place, instead of a hidden private data silo.

hbeni commented:
> PS: https://github.com/Supermagnum/fgcom-mumble/blob/master/docs/summary-of-changes.md

Thank you, just read it.

But why are the HF an String radio models be removed?
Why has the fgcom-mumble.ini been moved „centrally“ - it solely belongs to the mumble plugin.
What is the ATIS server component, how does it work? Why was it neccesary, ATIS recording and playback was already there and functional?

hbeni commented:
Thank you.

Is it possible to create that feature in a separate branch, based on current `master`?
This way I can apply it without cherry-picking, and also fine tuning it until its ready to merge is way easier.

Supermagnum commented:
> > > Thank you.
> > > Is it possible to create that feature in a separate branch, based on current `master`? This way I can apply it without cherry-picking, and also fine tuning it until its ready to merge is way easier.
> > 
> > 
> > Its already done, in Master. But,- I could push it to dev.
> 
> Great, then just file a separate PR for the feature 👍

Maybe, but master now has:
Added addNoiseSquelch config field (default: true) to control squelch noise
- Added AUDIO_FX_NOISES UDP protocol field for runtime control
- Added useLocationBasedNoise config flag to toggle location-based noise calculations
- Updated FGFS addon: added audioNoiseSquelch field (default: 0 for FlightGear)
- Updated RadioGUI: added checkbox option for audioNoiseSquelch (default: true)
- Added debug logging for addNoiseSquelch state
- Updated documentation: plugin.spec.md and FLIGHTGEAR_LIGHTNING_DATA.md
- FlightGear creates its own noise, so the plugin noise can be disabled via AUDIO_FX_NOISES=0

hbeni commented:
> but master now has

The problem is, that I cant cherry pick the feature, because it is based on altered codebase.
And `master` is not mergeable due to lacking testing/codeverification.

Hence my pledge:
- make a new feature branch based on upsream (my) master branch
- implement (port?) the feature to that branch

the result is a concise branch just containing that feature. It is probably small enough so I can find time to test and verify. And finally merge.

After that, you can rebase _this_ PR here to the new codebase (my master with your change merged)

hbeni commented:
Closed as per https://github.com/hbeni/fgcom-mumble/pull/240#issuecomment-3582235597
